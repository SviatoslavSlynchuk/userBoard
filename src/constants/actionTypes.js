export const SET_PERSONNEL_DATA = 'SET_PERSONNEL_DATA';
export const GET_PERSONNEL_DATA = 'GET_PERSONNEL_DATA';
export const SET_NEW_STATUS = 'SET_NEW_STATUS';

export const SET_FILTER_OPERATION = 'SET_FILTER_OPERATION';
export const SET_FILTER_VALUE = 'SET_FILTER_VALUE';

export const SET_STATE_FROM_LOCAL_STORAGE = 'SET_STATE_FROM_LOCAL_STORAGE';
