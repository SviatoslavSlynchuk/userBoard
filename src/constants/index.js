export const DATA_URL = 'https://randomuser.me/api/?nat=gb&results=5';
export const APPLIED = 'applied';
export const INTERVIEWING = 'interviewing';
export const HIRED = 'hired';

export const FILTER_TYPE_NAME = 'name';
export const FILTER_TYPE_CITY = 'city';
