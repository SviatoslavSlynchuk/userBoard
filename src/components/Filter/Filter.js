import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import * as constants from '../../constants';
import Helpers from '../../helpers';

const StyledSelect = styled.select`
    margin: 10px;
`;

const StyledInput = styled.input`
    margin: 10px;
`;

const StyledSpan = styled.span`
    margin: 10px;
`;

const Filter = props => (
    <div>
        <StyledSpan>Filter by:</StyledSpan>
        <StyledSelect
            value={props.filter.operation}
            onChange={props.onChangeFilterOperation}
        >
            <option value={constants.FILTER_TYPE_NAME}>
                {Helpers.capitalize(constants.FILTER_TYPE_NAME)}
            </option>
            <option value={constants.FILTER_TYPE_CITY}>
                {Helpers.capitalize(constants.FILTER_TYPE_CITY)}
            </option>
        </StyledSelect>
        <StyledInput
            type='text'
            value={props.filter.value}
            onChange={props.onChangeFilterValue}
        />
    </div>
);

Filter.propTypes = {
    filter: PropTypes.shape({
        operation: PropTypes.string.isRequired,
        value: PropTypes.string.isRequired,
    }),
    onChangeFilterOperation: PropTypes.func.isRequired,
    onChangeFilterValue: PropTypes.func.isRequired,
};

export default Filter;
