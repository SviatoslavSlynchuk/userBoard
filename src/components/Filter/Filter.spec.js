import React from 'react';
import Filter from './Filter';
import renderer from 'react-test-renderer';
import * as constants from '../../constants';

describe('Filter', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<Filter
                filter={{
                    operation: constants.FILTER_TYPE_NAME,
                    value: 'test',
                }}
                onChangeFilterOperation={() => {}}
                onChangeFilterValue={() => {}}
            />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
