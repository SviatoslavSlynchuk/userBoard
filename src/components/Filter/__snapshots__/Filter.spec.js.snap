// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter renders correctly 1`] = `
<div>
  <span
    className="sc-htpNat jTDBgV"
  >
    Filter by:
  </span>
  <select
    className="sc-bdVaJa kzpADP"
    onChange={[Function]}
  >
    <option
      value="name"
    >
      Name
    </option>
    <option
      value="city"
    >
      City
    </option>
  </select>
  <input
    className="sc-bwzfXH kxyJUq"
    onChange={[Function]}
    type="text"
  />
</div>
`;
